web/src/unread_ops.js

```diff
@@ -484,6 +484,22 @@ export function initialize() {
             // Update many places on the DOM to reflect unread
             // counts.
             process_visible();
+
+            // If messages have been sent to the current, visible
+            // message feed, check if the currently selected
+            // message is still unread after process_visible.
+            // If it is, select the message again in case it
+            // can be marked as read now that the window is in
+            // focus.
+            if (
+                narrow_state.is_message_feed_visible() &&
+                message_lists.current.selected_id() !== -1
```

Comment:
For this case, where we want the selected message in the message list view potentially marked as read, I didn't add a check for an open modal, which happens in `process_visible`.

My thought process was that we want the selected message marked as read (if possible) when the modal is closed.

If the user closes the tab, without looking at it, then nothing is marked as read. If they user clicks into the tab, but closes the tab without closing the modal, then they could potentially miss that the selected message was marked as read.

Reply:
This comment is out of date with the current version of these changes. I decided that it was better to be consistent with `process_visible` and not mark as read unless `viewport_is_visible_and_focused` returns true.

That way there is a bug that is consistent with both cases (bottom of the feed is visible, bottom of the feed isn't visible) where a overlay or modal is open in an unfocused window and new messages are received in the view.